TFT/src/User/API/Temperature.c

```diff
@@ -15,17 +15,25 @@ static bool    heat_send_waiting[MAX_HEATER_COUNT];
 uint32_t nextHeatCheckTime = 0;
 #define AUTOREPORT_TIMEOUT (nextHeatCheckTime + 3000)  // update interval + 3 second grace period
 
-static uint8_t fixHeaterIndex(uint8_t index)
+// Verify the index is invalid
+static bool heaterIndexInvalid(uint8_t index)
 {
-  if (infoSettings.hotend_count == 1)
-    index = (index < MAX_HOTEND_COUNT) ? NOZZLE0 : index;
-  return index;
```

Comment:
The reason for this check was to convert the index to `NOZZLE0` when the selected hotend index is greater than `NOZLE0` but less than the maximum possible hotend count. 
This was done to ensure the correct hotend index is selected in case of a mixing hotend which have multiple extruders with a single hotend because all the menu related to the extruder pass the extruder index to the temperature API for queries.
The logic only works when the hotend count is 1 and the selected hotend index is not `0`. It does not affect the heated bed index or the chamber heater index. If the hotend count is greater than 1, then there is no effect.